serde_codegen_internals/src/attr.rs

```diff
@@ -270,6 +278,17 @@ impl Item {
                         }
                     }
 
+                    // Parse `#[serde(types(from = "type", into = "type", as = "type"))]
+                    MetaItem(List(ref name, ref meta_items)) if name == "types" => {
```

Comment:
I am not worried about polluting the namespace (not yet anyway!). I think it will be fine as long as we group these in a sensible way in our documentation and examples. For example:

```rust
#[serde(as = "&str", from = "String")]
#[serde(rename = "myStr")]

// rather than
#[serde(as = "&str")]
#[serde(rename = "myStr")]
#[serde(from = "String")]
```